src/item_location.cpp

```diff
@@ -31,7 +33,14 @@ class item_location::impl
         virtual void remove_item() = 0;
 
     protected:
-        item *what = nullptr;
+        /** prevent auto-incrementation of the uuid via move construction */
+        impl( uuid &&uid ) : uid( std::move( uid ) ) {}
```

Comment:
This is a perfect example, the only reason you need to do a std::move() here is you made the uid copy constructor break its contract.